src/tracker/tracks_manager.py

```diff
@@ -0,0 +1,98 @@
+from src.tracker.track import Track
+from typing import Dict
+import sqlite3
+import numpy as np
+from src.config.config import TracksManagerConfig
+
+# TODO: make it more maintanable and have map[name_of_table, list[columns]]
+tables: Dict[str, str] = {
+    "tracks": """
+        CREATE TABLE IF NOT EXISTS tracks(
+            id CHAR PRIMARY KEY,
+            bbox BLOB,
+            embedding BLOB
+        )
+    """,
+    "category_count": """
+        CREATE TABLE IF NOT EXISTS category_count(
+            id CHAR PRIMARY KEY,
+            count INTEGER
+        )
+    """,
+}
+
+
+class TracksManager:
+    # TODO: keep track of the updated tracks since last sync so we don't serialize and write everything
+    def __init__(self, config: TracksManagerConfig):
+        self.tracks_on_disk: Dict[str, Track] = {}
+        self.category_count_on_disk: Dict[str, int] = {}
+        self.path_to_database = config.path_to_db.value
+        self.save_period = config.save_period
+        self.con: sqlite3.Connection = sqlite3.connect(self.path_to_database)
+        instantiate_table("tracks", self.con)
+        instantiate_table("category_count", self.con)
+
+    def close(self):
+        self.con.close()
+
+    def parse_tracks_on_disk(self):
+        # Fetch all track records from the database, including bbox
+        res = self.con.execute("SELECT * FROM tracks").fetchall()
+
+        for serialized_track in res:
+            track = self.get_new_track_from_bytes(serialized_track)
+            self.tracks_on_disk[track.track_id] = track
+
+    @staticmethod
+    def get_new_track_from_bytes(serialized_track: tuple):
+        assert len(serialized_track) == 3
```

Comment:
rather than an assert statement, create an if statement, and throw an exception with an error